visidata/loaders/pandas_freqtbl.py

```diff
@@ -0,0 +1,155 @@
+    def __init__(self, df, mask_iloc, index=0):
+        self.df = df
+        self.mask_iloc = mask_iloc
+        self.index = index
+
+    def __next__(self):
+        # Accessing row of original dataframe, to ensure
+        # that no copies are made and id() of selected rows
+        # will match original dataframe's rows
+        if self.index >= self.mask_iloc.shape[0]:
+            raise StopIteration()
+        row = self.df.iloc[self.mask_iloc[self.index]]
+        self.index += 1
+        return row
+
+class PandasSheetFreqTable(SheetPivot):
+    'Generate frequency-table sheet on currently selected column.'
+    rowtype = 'bins'  # rowdef FreqRow(keys, sourcerows)
+
+    def __init__(self, sheet, *groupByCols):
+        fqcolname = '%s_%s_freq' % (sheet.name, '-'.join(col.name for col in groupByCols))
+        super().__init__(fqcolname, groupByCols, [], source=sheet)
+        self.largest = 1
+
+    def selectRow(self, row):
+        # raise ValueError(str(type(self.source)) + " " + str(type(row.sourcerows)))
+        self.source.select(row.sourcerows)     # select all entries in the bin on the source sheet
+        return super().selectRow(row)  # then select the bin itself on this sheet
+
+    def unselectRow(self, row):
+        self.source.unselect(row.sourcerows)
+        return super().unselectRow(row)
+
+    def updateLargest(self, grouprow):
+        self.largest = max(self.largest, len(grouprow.sourcerows))
+
+    # @asyncthread
+    def reload(self):
+        'Generate frequency table then reverse-sort by length.'
+        # Conditions:
+        # (1) this only works if len(groupByCols) == 1 since it relies on value_counts()
+        # (2) this assumes some amount of non-degeneracy (non-empty dataframe with non-null values)
+        super().initCols(use_range=False)
+        # columns now the same as the original table
+        # raise ValueError(str([c.name for c in self.columns]))
+
+        df = self.source.rows
+
+        if len(self.groupByCols) > 1:
+            raise NotImplementedError("Frequency Table for > 1 columns currently not supported for PandasSheet")
+        # TODO: does this make a copy?
+        # raise ValueError(str(type(self.groupByCols[0])) + " " + str(self.groupByCols[0].name))
+        this_column = df.loc[:, str(self.groupByCols[0].name)]
+        value_counts = this_column.value_counts()
+
+        # add default bonus columns
+        for c in [
+                    Column('count', type=int, getter=lambda col,row: len(row.sourcerows)),
+                    Column('percent', type=float, getter=lambda col,row: len(row.sourcerows)*100/df.shape[0]),
+                    Column('histogram', type=str, getter=lambda col,row: options.disp_histogram*(options.disp_histolen*len(row.sourcerows)//value_counts.max()), width=options.disp_histolen+2),
+                    ]:
+            self.addColumn(c)
+
+        # TODO: can make this part async w/ progress bar
+        for element in value_counts.index:
```

Comment:
if `.index` has a len(), then you can use `Progress(value_counts.index)` and that will take care of it.  Otherwise `with Progress(total=n) as prog: ... prog.addProgress(1)`